sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java

```diff
@@ -473,56 +475,72 @@ public OutputT defaultValue() {
     @Override
     public <K> KeyedCombineFn<K, InputT, AccumT, OutputT> asKeyedFn() {
       // The key, an object, is never even looked at.
-      return new KeyedCombineFn<K, InputT, AccumT, OutputT>() {
-        @Override
-        public AccumT createAccumulator(K key) {
-          return CombineFn.this.createAccumulator();
-        }
+      return new KeyIgnoringCombineFn<>(this);
+    }
 
-        @Override
-        public AccumT addInput(K key, AccumT accumulator, InputT input) {
-          return CombineFn.this.addInput(accumulator, input);
-        }
+    private static class KeyIgnoringCombineFn<K, InputT, AccumT, OutputT>
```

Comment:
:+1: to this anyhow. More readable. It highlights how annoying marker/mixin interfaces can be, though. If we already had the annotation-driven machinery for `CombineFn` I would suggest using it. But since we don't have it yet, I think this is fine.